Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -850,6 +854,13 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
     return .visitChildren
   }
 
+  override func visitPost(_ node: MemberAccessExprSyntax) {
+    let nodeID = node.id
+    if rootExprs.remove(nodeID) != nil {
+      preVisitedExprs.removeAll()
+    }
```

Comment:
Can you pull this into a private helper function that takes a generic argument `<Node: ExprSyntaxProtocol>`? Then each of these `visitPost`s will be a one-liner.

Reply:
Done